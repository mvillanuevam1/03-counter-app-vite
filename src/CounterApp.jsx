import { useState } from 'react'
import PropTypes from "prop-types";



export const CounterApp = (  {value}  ) => {


    const [ counter, setCounter ] = useState( value );


    const handleAdd = () => {
        setCounter( counter + 1 )
    }

    const handleSub = () => {
        setCounter( counter - 1 ) 
    }

    const handleReset = () => {
        setCounter( value ) 
    }

    return (
        <>
            <h1>CounterApp</h1>
            <button onClick={ handleAdd } >+1</button>
            <button onClick={ handleSub } >-1</button>
            <button aria-label="btn-reset" onClick={ handleReset } >Reset</button>
            <h2> { counter } </h2>
        </>
    );

}



CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 'No hay un valor inicial'
}

