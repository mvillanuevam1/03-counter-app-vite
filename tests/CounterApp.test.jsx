
import { fireEvent, render, screen } from "@testing-library/react";
import {CounterApp} from "../src/CounterApp.jsx";



describe('Pruebas en CounterApp.jsx', () => {

    const value = 10;

    test('debe hacer match con el snapshot', () => {

        const {container} = render(<CounterApp

            value={value}
        
        />);
    
        expect(container).toMatchSnapshot();
    

    });

    test('el value que se muestra debe ser 100', () => {

        const {get} = render(<CounterApp

            value={value}

        />);


        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('10');

    });


    test('debe incrementar con el boton +1', () => {

        render(<CounterApp

            value={value}

        />);

        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('11')).toBeTruthy();

    });
    

    test('debe decrementar con el boton -1', () => {

        render(<CounterApp

            value={value}

        />);

        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('9')).toBeTruthy();

    });


    test('debe de funcionar el boton de reset', () => {

        render(<CounterApp

            value={value}

        />);

        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        // fireEvent.click(screen.getByText('Reset'));

        fireEvent.click(screen.getByRole('button', {name:'btn-reset'}));

        expect(screen.getByText('10')).toBeTruthy();

    });



});

