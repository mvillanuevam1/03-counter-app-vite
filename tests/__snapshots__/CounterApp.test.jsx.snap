// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp.jsx debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <h2>
     
    10
     
  </h2>
</div>
`;
